flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/LogicalCorrelateToJoinFromTemporalTableFunctionRule.scala

```diff
@@ -153,9 +153,8 @@ class LogicalCorrelateToJoinFromTemporalTableFunctionRule
     */
   private def getRelOptSchema(relNode: RelNode): RelOptSchema = relNode match {
```

Comment:
we should avoid to get `RelOptSchema` from RelNode, another approach is we can add a method to get `FlinkRelBuilder` directly in `FlinkOptimizeContext`.

Reply:
Hi @godfreyhe, I commit the code of this issue, Please help to review.